packages/python/plotly/plotly/express/_core.py

```diff
@@ -821,6 +827,11 @@ def build_or_augment_dataframe(args, attrables, array_attrables):
                         "Expected one of %s but received: %s"
                         % (field, str(list(df_columns)), argument)
                     )
+                except TypeError:
```

Comment:
This seems overly-broad to me, compared to just directly checking if the `data_frame` argument has been passed in... I worry about try-catch control flow as you never know what new crazy thing will raise any given exception later :)

Reply:
ok, I hesitated indeed between the two solutions. I chose the try/except mostly for the sake of compactness since this function is getting crazily long but this is not such a good argument compared to robustness.